File: oop-fullstack-lab/caneda/frontend/src/types/CoinFlip.ts
export interface CoinFlip {
  id: number;
  userChoice: string;
  outCome: string;
  winOrNo: boolean;
  timeFlippedAt: string;
  user: {
    id: number;
    name: string;
    email: string;
    role: string;
  };
}

File: oop-fullstack-lab/caneda/backend/main/java/com/caneda/fullstack/entity/CoinFlip.java
package com.caneda.fullstack.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data // generates getters, setters, toString, equals, hashCode
@Entity // JPA entity maps to a database table
@Table(name = "CoinFlips") // Name of the table in PostgreSQL
public class CoinFlip {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String userChoice;
    private String outCome;
    private boolean winOrNo;
    private LocalDateTime timeFlippedAt;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;



}
